pkg/talos/networkconfig.go

```diff
@@ -93,6 +98,143 @@ func GenerateNetworkConfigBytes(ifCfg *config.IngressFirewall) ([]byte, error) {
 	return CombineYamlBytes(result), nil
 }
 
+func GenerateLinkAliasConfigBytes(devices []*v1alpha1.Device) ([]byte, error) {
+	var result [][]byte
+
+	for _, device := range devices {
+		if device.DeviceSelector == nil {
+			continue
+		}
+
+		aliasConfig := GenerateLinkAliasConfig(device)
+		if aliasConfig == nil {
+			continue
+		}
+
+		aliasBytes, err := marshalYaml(aliasConfig)
+		if err != nil {
+			return nil, err
+		}
+
+		result = append(result, aliasBytes)
+	}
+
+	if len(result) == 0 {
+		return nil, nil
+	}
+
+	return CombineYamlBytes(result), nil
+}
+
+func GenerateLinkAliasConfig(device *v1alpha1.Device) *network.LinkAliasConfigV1Alpha1 {
+	if device == nil || device.DeviceSelector == nil {
+		return nil
+	}
+
+	aliasConfig := network.NewLinkAliasConfigV1Alpha1(device.DeviceInterface)
```

Comment:
`device.DeviceInterface` and `device.DeviceSelector` are mutually exclusive afaik. It will error out when both are set in <v1.12.

So, `device.DeviceInterface` will always be empty string if `device.DeviceSelector` is set. I think we should give it a name manually like `iface[0-9]`?